std/int128.d

```diff
@@ -263,13 +263,19 @@ public struct Int128
         return this == op2 ? 0 : gt(this.data, op2.data) * 2 - 1;
     }
 
-    /** support signed arithmentic comparison operators < <= > >=
+    /** support arithmentic comparison operators < <= > >=
      * Params: op2 = right hand operand
      * Returns: -1 for less than, 0 for equals, 1 for greater than
      */
-    int opCmp(long op2) const
+    int opCmp(Long)(const Long op2) const
+    if (is(Long : long))
     {
-        return opCmp(Int128(0, op2));
+        static if (!__traits(isIntegral, Long)) // Long converts to integer through alias this.
+            return opCmp(__traits(getMember, op2, __traits(getAliasThis, Long)[0]));
```

Comment:
I think it's covered by the template constraint `is(Long : long)`.

Reply:
My bigger concern is whether we need to add code to support `alias this` in the first place. It reeks a bit of Phobos' generality creep. It's also not part of the issue, so it could use its own separate PR.

Reply:
[That is to say, this code currently works so breaking it would be a regression:](https://run.dlang.io/is/BVoXVA)

```d
struct Wrapper
{
    long value;;
    alias value this;
}
void main()
{
    import std.int128 : Int128;
    assert(Int128(1L) < Wrapper(2L));
}
```